plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -3756,6 +3760,42 @@ public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(
         commitUpdateAndTransaction(appendFiles, session, transaction, "refresh materialized view");
         transaction = null;
         fromSnapshotForRefresh = Optional.empty();
+
+        // cleanup old snapshot
+        if (snapshotIdBeforeRefreshCommit != null) {
```

Comment:
Does this logic work well on pre-existing materialized views having more than one snapshot?